static/src/js/containers/DownloadHistoryContainer/__tests__/DownloadHistoryContainer.test.js

```diff
@@ -1,89 +1,250 @@
 import React from 'react'
-import Enzyme, { shallow } from 'enzyme'
-import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
-
-import actions from '../../../actions'
 import {
-  DownloadHistoryContainer,
-  mapDispatchToProps,
-  mapStateToProps
-} from '../DownloadHistoryContainer'
+  render,
+  waitFor,
+  act
+} from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { Provider } from 'react-redux'
+import configureStore from 'redux-mock-store'
+import nock from 'nock'
+
+import { addToast } from '../../../util/addToast'
+import { handleError } from '../../../actions/errors'
+import DownloadHistoryContainer from '../DownloadHistoryContainer'
 import { DownloadHistory } from '../../../components/DownloadHistory/DownloadHistory'
 
-Enzyme.configure({ adapter: new Adapter() })
+jest.mock('../../../util/addToast', () => ({
+  addToast: jest.fn()
+}))
+
+jest.mock('../../../actions/errors', () => ({
+  handleError: jest.fn(() => ({ type: 'HANDLE_ERROR' }))
+}))
+
+let lastPassedOnDeleteRetrieval = null
```

Comment:
Instead of using this variable to track the last retrieval, you can import `DownloadHistory` which will be your mocked component and then use `toHaveBeenCalledTimes` and `toHaveBeenCalledWith` to determine if its called the right number of times and with the correct values.

Reply:
Scratch that, kinda. I needed to take a deeper look a this but I think I see whats going on. I think rather than using this variable to track the most recent call, the more "testing-library" way to do this would be to return a `button` in the mocked component that has an `onClick` which calls `onDeleteRetrieval`. Then in your tests, you can use `userEvent` to click on the button and then call `toHaveBeenCalledWith()` to make sure `onDeleteRetrieval` is called with what we expect.